src/cli.ts

```diff
@@ -178,19 +180,31 @@ function startTestRunner(type: JestMode, args: CLITestArguments) {
  * Starts the build + server process, passing in any provided CLI arguments
  */
 async function startCommandProcess(args: CLIArguments) {
-    const sanitizedArgs: CLIArguments = {
-        providerVersion: args.providerVersion || 'local',
-        mode: args.mode || 'development',
-        noDemo: args.noDemo === undefined ? false : true,
-        static: args.static === undefined ? false : true,
-        writeToDisk: args.writeToDisk === undefined ? false : true,
-        runtime: args.runtime
+    const parsedArgs: CLIArguments = {
+        providerVersion: 'local',
+        mode: 'development',
+        noDemo: false,
+        static: false,
+        writeToDisk: false,
+        runtime: undefined,
+
+        // Hooks can selectively override the above defaults. CLI args will still take precedence.
+        ...allowHook(Hooks.DEFAULT_ARGS, {})()
     };
 
+    // Apply CLI-specified arguments on top of defaults
+    const argList = Object.keys(parsedArgs) as (keyof CLIArguments)[];
+    argList.forEach(<K extends keyof CLIArguments>(key: K) => {
+        if (args.hasOwnProperty(key)) {
+            parsedArgs[key] = args[key];
+        }
+    });
+
     const server = await createServer();
-    await createDefaultMiddleware(server, sanitizedArgs);
+    allowHook(Hooks.APP_MIDDLEWARE)(server);
+    await createDefaultMiddleware(server, parsedArgs);
```

Comment:
That's what the `APP_MIDDLEWARE` hook is doing - allowing apps to register their own middleware that takes precedence over the built-in middleware.

I don't think we should remove the default middleware and have each service/app provide their own (if that's what you're suggesting?) since that would result in duplication.

Reply:
Not what I'm suggesting. I mean could we make this work like a hook internally instead of some special case

Reply:
Discussed offline. Will mark resolved